app/commands/v2/publish.rb

```diff
@@ -39,6 +39,7 @@ def call
         clear_published_items_of_same_locale_and_base_path(content_item, translation, location)
 
         set_public_updated_at(content_item, previous_item, update_type)
+        set_first_published_at(content_item)
```

Comment:
Not the fault of this line specifically, but this method is getting really long.